build/scripts/DevEnv/ALDev.psm1

```diff
@@ -0,0 +1,225 @@
+Import-Module "$PSScriptRoot\..\EnlistmentHelperFunctions.psm1" -DisableNameChecking
+
+function GetAssemblyProbingPaths($ContainerName) {
+    $netPackages = @()
+    $netPackages += @(Get-ChildItem -Path "$($bcContainerHelperConfig.containerHelperFolder)\\Extensions\\$ContainerName\\.netPackages\\Shared\\Microsoft.AspNetCore.App" | Sort-Object -Descending | Select-Object -ExpandProperty FullName)
+    $netPackages += @(Get-ChildItem -Path "$($bcContainerHelperConfig.containerHelperFolder)\\Extensions\\$ContainerName\\.netPackages\\Shared\\Microsoft.NETCore.App" | Sort-Object -Descending | Select-Object -ExpandProperty FullName)
+    $netPackages += "$($bcContainerHelperConfig.containerHelperFolder)\\Extensions\\$ContainerName\\.netPackages\\Service"
+    $netPackages += "$($bcContainerHelperConfig.containerHelperFolder)\\Extensions\\$ContainerName\\.netPackages"
+
+    $assemblyProbingPaths = $netPackages | ForEach-Object { return ($_ -replace '\\', '/') }
+    return $assemblyProbingPaths
+}
+
+function GetDefaultProjectSettings($ContainerName)
+{
+    $defaultSettings = GetDefaultSettings -SettingsJson
+    $defaultSettings["al.assemblyProbingPaths"] = (GetAssemblyProbingPaths -ContainerName $ContainerName)
+    $defaultSettings["al.packageCachePath"] = (Get-ArtifactsCacheFolder -ContainerName $ContainerName)
+
+    # Output the compiled app files into the artifacts cache folder
+    $compilationOptions = @{
+        "outFolder" = (Get-ArtifactsCacheFolder -ContainerName $ContainerName)
+    }
+    $defaultSettings["al.compilationOptions"] = $compilationOptions
+
+    return $defaultSettings
+}
+
+function MergeSettings($CustomLaunchSettings, $DefaultLaunchSettings) {
+    foreach ($propertyName in $DefaultLaunchSettings.Keys)
+    {
+        if (!$CustomLaunchSettings.ContainsKey($propertyName))
+        {
+            $CustomLaunchSettings[$propertyName] = $DefaultLaunchSettings[$propertyName]
+        }
+    }
+    return $CustomLaunchSettings
+}
+
+function GetDefaultSettings([switch]$LaunchJson, [switch]$SettingsJson)
+{
+    if ($LaunchJson)
```

Comment:
There is also an option that both switch would be provided, not sure if that's breaking or sth ;)

Reply:
Yeah that's true. Wouldn't really make sense though 😄 We only use this internally so for now I think it's fine